packages/screenreaderonly/src/react/index.tsx

```diff
@@ -0,0 +1,18 @@
+import { accessibility } from '@pluralsight/ps-design-system-core'
+import { css } from 'glamor'
+import React, { ReactNode, forwardRef } from 'react'
+
+interface Props {
+  children?: ReactNode
+  as?: React.ElementType | keyof JSX.IntrinsicElements
+}
+
+const ScreenReaderOnly = forwardRef<any, Props>((props, ref) => {
+  const { as: Tag = 'div', ...rest } = props
+
+  return <Tag ref={ref} {...css(accessibility.screenReaderOnly)} {...rest} />
+})
```

Comment:
some comments.

- typescript makes it really hard to get true type safety with the `as` polymorphic prop. srsly...i went down some rabbit holes. i just gave up
- `forwardRef` does not have a good way to do generics or overloading
- a better solution in the future for the polymorphic `as` component is not to use it...but instead use a render prop https://blog.andrewbran.ch/polymorphic-react-components/#an-alternative-approach

Reply:
I'm fine to move the render prop approach now or later.